initializr-generator-spring/src/main/java/io/spring/initializr/generator/spring/container/dockercompose/DockerComposeProjectContributor.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2012-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.spring.initializr.generator.spring.container.dockercompose;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import io.spring.initializr.generator.project.contributor.ProjectContributor;
+
+/**
+ * A {@link ProjectContributor} which contributes a 'compose.yaml' file through a
+ * {@link DockerComposeFile}. The file only gets created when the
+ * {@link DockerComposeFile#isEmpty() docker compose file isn't empty}.
+ *
+ * @author Moritz Halbritter
+ */
+class DockerComposeProjectContributor implements ProjectContributor {
+
+	private final DockerComposeFile composeFile;
+
+	DockerComposeProjectContributor(DockerComposeFile composeFile) {
+		this.composeFile = composeFile;
+	}
+
+	@Override
+	public void contribute(Path projectRoot) throws IOException {
+		if (this.composeFile.isEmpty()) {
+			return;
```

Comment:
I am not sure about that. Is a yaml with an empty services tag a valid document? 

I am asking as if you add docker-compose without any matching service, then you won't create the file and the app won't start. I wonder what's best here, preparing an empty document (considering it is valid) and perhaps have a note in HELP.md that state no service were added.

Reply:
Unfortunately, `compose.yaml` with an empty `services:` in it is not valid:

```
╰─$ docker compose up
validating /home/moe/tmp/docker-compose-test/compose.yaml: services must be a mapping
```

So if we create no file, the app won't start. And if we create an empty file, the app won't start either.

Reply:
Hum. I don't know what to do. Flagging for team attention.

Reply:
We now add warnings if no services are defined. The `compose.yaml` file gets written, but only has `services:` in it.